pkg/actuators/awsplacementgroup/controller.go

```diff
@@ -0,0 +1,103 @@
+package awsplacementgroup
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	machinev1 "github.com/openshift/api/machine/v1"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	mapierrors "github.com/openshift/machine-api-operator/pkg/controller/machine"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"k8s.io/klog/v2"
+)
+
+// Reconciler reconciles machineSets.
+type Reconciler struct {
+	Client client.Client
+	Log    logr.Logger
+
+	recorder record.EventRecorder
+	scheme   *runtime.Scheme
+}
+
+// SetupWithManager creates a new controller for a manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	_, err := ctrl.NewControllerManagedBy(mgr).
+		For(&machinev1.AWSPlacementGroup{}).
+		WithOptions(options).
+		Build(r)
+
+	if err != nil {
+		return fmt.Errorf("failed setting up with a controller manager: %w", err)
+	}
+
+	r.recorder = mgr.GetEventRecorderFor("awsplacementgroup-controller")
+	r.scheme = mgr.GetScheme()
+	return nil
+}
+
+// Reconcile implements controller runtime Reconciler interface.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("awsplacementgroup", req.Name, "namespace", req.Namespace)
+	logger.V(3).Info("Reconciling")
+
+	awsPlacementGroup := &machinev1.AWSPlacementGroup{}
+	if err := r.Client.Get(ctx, req.NamespacedName, awsPlacementGroup); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Object not found, return. Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Ignore deleted AWSPlacementGroups, this can happen when foregroundDeletion
+	// is enabled
+	if !awsPlacementGroup.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+	originalAWSPlacementGroupToPatch := client.MergeFrom(awsPlacementGroup.DeepCopy())
```

Comment:
Will this handle status as well?

Reply:
It looks like `func (in *AWSPlacementGroup) DeepCopy() ` does copy both `.Spec` and `.Status`
https://github.com/openshift/api/blob/master/machine/v1/zz_generated.deepcopy.go#L30-L48
So when we `.Patch()` it should be for both, right?

Reply:
Very much depends whether the patch call is going to call status.Patch as well as just Patch, will need to inspect the code to see what it's doing under the hood, else a status subresource might be left behind and not updated

Reply:
Having now reviewed the other code, no, we aren't patching the status at the moment, the `Client.Patch` won't update the status, you will need to use the status client separately to do that

Reply:
Yes, came to the same conclusion. Thanks for flagging!